pureftpd/symlink check/check_symlink_accounts.sh

```diff
@@ -0,0 +1,20 @@
+#! /bin/bash
+MYSQL=/usr/bin/mysql
+LOGFILE=/var/log/symlink.log
+TS=`date '+%Y-%m-%d %H:%M:%S'`
+sylinks=0
+$MYSQL --defaults-file=/etc/mysql/debian.cnf -N -e "use pureftpd; select Dir from users;" | \
+(while read dir; do
+        dir1=`echo "$dir" | sed 's/\/$//'`
+        if [ -L "${dir1}" ] && [ -e ${dir1} ] ; then
+                TS=`date '+%Y-%m-%d %H:%M:%S'`
+                echo "$TS $dir1 is symlink" >> $LOGFILE
+                ((sylinks++))
+        fi
+done
+
+if [[ $sylinks -gt 0 ]]
+then
+        echo "CRITICAL - Number of ftp accounts with symlinks as root is $sylinks" >> $LOGFILE
+fi
```

Comment:
In case you had CRITICAL state in the log file `/var/log/symlink.log` and it recovered to OK, you will still have CRITICAL in the log.

I suggest having two files:
1. The lock file that will have CRITICAL or OK at one moment i.e. `/var/log/symlink.lock`
2. The next log that will check if CRITICAL accounts are found can be `/var/log/symlink.log`

On line 18, you could log CRITICAL only to the lock file with `>` so the lock will get overridden.

Issue with this would be if, an attacker was fast enough and Nagios does not alert for 1 CRITICAL. It can be configured differently in Nagios.